### PR TITLE
[BREAKING] Chore/prerelease refactor

### PR DESCRIPTION
**Pull request by vmidyllic**
the new version of auth library:
1. FullVerify
2. VerifyJWZ
3. Verify - auth messages.

Pub signals verification specific implementation for AtomicMtp and auth circuits

### CODE REVIEW

**Review by AndriianChestnykh — COMMENTED** [review]
A few comments for now. Didn't reviewed deeply the verification package.

File: auth.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	if err != nil {
+		return nil, err
+	}
+	var cv pubsignals.Verifier
+	switch circuitID {
+	case circuits.AtomicQueryMTPCircuitID:
+		cv = &pubsignals.AtomicQueryMTP{}
+	case circuits.AuthCircuitID:
+		cv = &pubsignals.Auth{}
+	}
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply auth.go]: reworked.

File: auth.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		return nil, err
+	}
+	verificationKey, err := circuits.GetVerificationKey(circuits.CircuitID(t.CircuitID))
+	if err != nil {
+		return nil, err
+	}
+	err = t.Verify(verificationKey)
+	if err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply auth.go:76]: will fix this after review of jwz library

File: go.mod [hunk 3]
```diff
@@ -4,34 +4,65 @@ go 1.18
 
 require (
 	github.com/ethereum/go-ethereum v1.10.17
+	github.com/gofrs/uuid v3.3.0+incompatible
```
Comment (olomix) [inline go.mod:7]: Why not more common github.com/google/uuid? At least it is go.mod-friendly.

File: pubsignals/signals.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+var signalsVerifierRegistry = map[circuits.CircuitID]reflect.Type{}
+var circuitsLock = new(sync.RWMutex)
+
+// ErrUserStateIsNotValid declares that issuer state is invalid
+var ErrUserStateIsNotValid = errors.New("user state is not valid")
+
+// ErrIssuerClaimStateIsNotValid declares that issuer state is invalid
+var ErrIssuerClaimStateIsNotValid = errors.New("issuer state is not valid")
```
Comment (olomix) [inline pubsignals/signals.go:17]: this error is never used.

**Review by olomix — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (vmidyllic) [reply go.mod]: fixed

**Review by demonsh — APPROVED** [review]

**Review by olomix — APPROVED** [review]

**Pull request merged**